src/js/components/AppListItemComponent.jsx

```diff
@@ -19,6 +21,71 @@ var AppListItemComponent = React.createClass({
     model: React.PropTypes.object.isRequired
   },
 
+  getInitialState: function () {
+    return {numVisibleLabels: -1};
+  },
+
+  componentDidMount: function () {
+    // Avoid referencing window from Node context
+    if (global.window != null) {
+      window.addEventListener("resize", this.handleResize);
+      window.addEventListener("focus", this.handleResize);
+    }
+    this.updateNumVisibleLabels();
+
+  },
+
+  componentDidUpdate: function (prevProps) {
+    if (!_.isEqual(this.props, prevProps)) {
```

Comment:
We try to get rid of underscore. Only place where it is used is in the tooltipMixin, which will have a refactoring too at some point.
You could introduce an isEqual() in the Util-collection.